documentation: mention additionalArgs (#6148)
prometheus-operator/prometheus-operator: Documentation/user-guides/strategic-merge-patch.md

    - sleep
    - "3600"
```

> ## How to inject additional cli arguments into the prometheus pods

Reviewer: (nit)
```suggestion
## How to inject additional CLI arguments into the prometheus container
```

> ## How to inject additional CLI arguments into the prometheus pods

Reviewer 2: ```suggestion
## How to inject additional CLI arguments into the prometheus container
```

> The following manifest injects an additional cli argument in the default

Reviewer: ```suggestion
The following manifest injects an additional CLI argument in the default
```

> Prometheus argument list. Not the use of `additionalArgs` in this example.

Reviewer 3: ```suggestion
Prometheus argument list. Note the use of `.spec.additionalArgs` in this example.
```

> Using `args` directly would overwrite the args list completely, including the

Reviewer 3: ```suggestion
Using `.spec.containers[*].args` instead would overwrite the container's args list completely, including the
```

> Using `.spec.containers[*].args` directly would instead overwrite the containers

Reviewer: ```suggestion
Using `.spec.containers[*].args` directly would instead overwrite the container's
```

```yaml
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: additional-arguments
spec:
  containers:
  - name: prometheus

>     additionalArgs:

Reviewer 3: I thought it was directly under `spec`?

>     - '--scrape.timestamp-tolerance=15ms'

Reviewer 3: maybe just `--arg=value`?

    - sleep
    - "3600"
```

## How to inject additional CLI arguments into the prometheus pods

The following manifest injects an additional CLI argument in the default
Prometheus argument list. Note the use of `.spec.additionalArgs` in this 
example.
Using `.spec.containers[*].args` directly would instead overwrite the containers 
`args` list completely, including the default arguments.

```yaml
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: additional-arguments
spec:
  containers:

>   - name: prometheus
>     additionalArgs:
>   - "--scrape.timestamp-tolerance=15ms"

Reviewer: ```suggestion
  additionalArgs:
  - name: "scrape.timestamp-tolerance"
  - value: "15ms"
```
